test/dynamo/test_functions.py

```diff
@@ -1835,6 +1835,87 @@ def fn(x):
         ref = opt_fn(x)
         self.assertEqual(ref, res)
 
+    def test_nested_inlined_graph_break(self):
+        def h(x):
+            x = x.cos()
+            print(x)
+            x = x.sin()
+            return x
+
+        def fn(x):
+            x = x.tan()
+            x = h(x)
+            x = x.tan()
+            return x
+
+        cnts = torch._dynamo.testing.CompileCounter()
+        opt_fn = torch._dynamo.optimize(backend=cnts)(fn)
+        x = torch.randn(4)
+        res = fn(x)
+        ref = opt_fn(x)
+        self.assertEqual(ref, res)
+
+        # First graph is x = cos(tan(x))
+        # Second graph is x = tan(sin(x))
+        self.assertEqual(cnts.frame_count, 2)
+
+    def test_nested_inlined_graph_break_2(self):
+        def h(x):
+            x = x.cos()
+            print(x)
+            x = x.sin()
+            return x
+
+        def fn(x):
+            x = x.tan()
+            x = h(x)
+            x = x.tan()
+            x = h(x)
+            x = x.tan()
+            return x
+
+        cnts = torch._dynamo.testing.CompileCounter()
+        opt_fn = torch._dynamo.optimize(backend=cnts)(fn)
+        x = torch.randn(4)
+        res = fn(x)
+        ref = opt_fn(x)
+        self.assertEqual(ref, res)
+
+        # First graph is x = cos(tan(x))
+        # Second graph is x = cos(tan(sin(x)))
+        # Third graph is x = tan(sin(x))
+        self.assertEqual(cnts.frame_count, 3)
+
+    def test_doubly_nested_inlined_graph_break(self):
```

Comment:
These tests should be improved 
- return different var names in each function
- return irregular number of args / have non-congruent number of local vars at time of graph break.